lib/practice_exercises.rb

```diff
@@ -1,13 +1,55 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) because we have an if loop inside an until loop, both depend on n
+# Space Complexity: O(n) because the size that the array takes up depends on its length, ie. n
 def is_palindrome(string)
```

Comment:
👍 However you don't have a loop inside of a loop, you have some loops in sequence.  So this is O(n) time and space complexity.